### PR TITLE
Back to empty user agent

### PR DESCRIPTION
**Pull request by alexarchambault**
Fixes https://github.com/coursier/coursier/issues/2287.

### CODE REVIEW

File: modules/cache/jvm/src/main/scala/coursier/cache/CacheUrl.scala [hunk 1]
```diff
@@ -140,6 +140,10 @@ object CacheUrl {
         // handling those ourselves, so that we can update credentials upon redirection
         conn0.setInstanceFollowRedirects(false)
 
+        // Early in the development of coursier, I ran into some repositories (Sonatype ones?) not
+        // returning the same content for user agent "Java/…".
+        conn0.setRequestProperty("User-Agent", "")
```
Comment (julienrf) [inline modules/cache/jvm/src/main/scala/coursier/cache/CacheUrl.scala:145]: Can we make

```suggestion
        conn0.setRequestProperty("User-Agent", "Coursier")
```

As initially proposed in #2246?

**Review by eed3si9n — APPROVED** [review]
Thanks 🙏

File: modules/cache/jvm/src/main/scala/coursier/cache/CacheUrl.scala (hunk 1 above)
Comment (eed3si9n) [reply modules/cache/jvm/src/main/scala/coursier/cache/CacheUrl.scala:145]: curl test says `"Coursier"` would work too

```bash
$ curl -H "User-Agent: Coursier" https://oss.sonatype.org/content/repositories/snapshots/org/zeromq/jeromq/0.5.3-SNAPSHOT/maven-metadata.xml
```

so I am happy with either.

Comment (alexarchambault) [reply modules/cache/jvm/src/main/scala/coursier/cache/CacheUrl.scala:145]: I think I'm going to cut a release with an empty user agent (former behavior), then one with a `Coursier` user agent.

**Pull request merged**